replacer.go

```diff
@@ -354,7 +354,9 @@ func (f fileReplacementProvider) replace(key string) (any, bool) {
 			zap.Error(err))
 		return nil, true
 	}
-	return string(body), true
+	content := string(body)
+	content = strings.TrimSuffix(content, "\n")
```

Comment:
Let's try this?

```suggestion
	content = strings.TrimRight(content, "\r\n")
```

Reply:
> Let's try this?

`TrimRight` would remove all trailing newlines, not just the last one.

I would suggest this:
```go
	// Remove a single trailing newline, regardless of its type
	if strings.HasSuffix(content, "\r\n") {
		return strings.TrimSuffix(content, "\r\n"), true
	}
	if strings.HasSuffix(content, "\n") {
		return strings.TrimSuffix(content, "\n"), true
	}
	return strings.TrimSuffix(content, "\r"), true
```

Reply:
Still failing... it's weird:
```
    replacer_test.go:445: Expected value 'foo
        ' for key 'file.caddytest/integration/testdata/foo_with_multiple_trailing_newlines.txt' got 'foo
        '
--- FAIL: TestReplacerNew (0.01s)
```

Reply:
The comparison in the testcase happens against "foo\n", but on the windows test it would need to compare against "foo\r\n".